controllers/apim/ratelimitpolicy_controller_test.go

```diff
@@ -0,0 +1,269 @@
+//go:build integration
+
+package apim
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	istioclientgoextensionv1alpha1 "istio.io/client-go/pkg/apis/extensions/v1alpha1"
+	istioclientnetworkingv1alpha3 "istio.io/client-go/pkg/apis/networking/v1alpha3"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	gatewayapiv1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+
+	apimv1alpha1 "github.com/kuadrant/kuadrant-controller/apis/apim/v1alpha1"
+	"github.com/kuadrant/kuadrant-controller/pkg/common"
+	"github.com/kuadrant/kuadrant-controller/pkg/rlptools"
+	limitadorv1alpha1 "github.com/kuadrant/limitador-operator/api/v1alpha1"
+)
+
+func testBuildBasicGateway(gwName, ns string) *gatewayapiv1alpha2.Gateway {
```

Comment:
Maybe we could name this one, and `testBuildBasicHttpRoute`, `testBuildBasicRoutePolicy`, etc... `fake` or even `mock`, since those are not testing but acting as a _double_ . I.E: `mockBasicHttpRoute`

Reply:
well, it is not a mock. It is just a builder method. I added `test` prefix to avoid collision with any other name from the same package. 

In go, regular test starts with `TestXXXX` so there is no issue with that.